students/charlie_england/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,122 @@
+import peewee
+import logging
+
+logging.basicConfig(filename='basic_operations.log', filemode='w', format='%(asctime)s - %(message)s', datefmt='%d-%b-%y %H:%M:%S',level='DEBUG')
+
+database = peewee.SqliteDatabase("customers.db")
+database.connect()
+database.execute_sql('drop table if exists customer;')
+
+class BaseModel(peewee.Model):
+    class Meta:
+        database = database
+
+class Customer(BaseModel):
+    """
+        This class defines Customer, which maintains details of someone
+        for whom we want to research consumer habits to date.
+
+        Customer ID
+        Name
+        Lastname
+        Home address
+        Phone Number
+        Email address
+        Status (active or inactive customer)
+        Credit Limit
+    """
+
+    customer_id = peewee.CharField(primary_key=True, max_length=30)
+    name = peewee.CharField(max_length=50)
+    lastname = peewee.CharField(max_length=50, null=True)
+    address = peewee.CharField(max_length=75)
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320)
+    status = peewee.CharField(max_length=10)
+    credit_limit = peewee.FloatField()
+
+database.create_tables([Customer])
+
+def add_customer(customer_id, name, lastname, address, phone_number, email, status, credit_limit):
+    """
+        adds a new customer to the database.
+    """
+    try:
+        with database.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+            )
+            customer.save()
+            logging.info(f'Customer {customer_id}/{name} created and added to database')
+    except Exception as e:
+        logging.error(f'Could not add {customer_id} due to {e}')
+
+def search_customer(customer_id):
+    """
+        searches for an active customer given a customer ID
+        returns a dictionary of the name, lastname, phone number and email
+        if customerID not found, returns an empty dictionary
+    """
+    try:
+        customer = Customer.get(Customer.customer_id == customer_id)
+        logging.info(f'Customer {customer_id} found, returning information')
+        return {
+            'name':customer.name,
+            'lastname':customer.lastname,
+            'phone_number':customer.phone_number,
+            'email':customer.email,
+        }
+    except Exception as e:
```

Comment:
Just as a note, you can use Customer.DoesNotExist as an exception here so that you don't have a catch all. This will better to use since you know exactly what the error is!

You will want to apply this for delete and update customer as well (since they first look for a customer via id)